### PR TITLE
Resolve a number of data races

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
Was working around with `./configure --with-sanitizers=thread` and found data races around this. These locks / added cses resolved the issues. It might be overly broad or not broad enough :shrug:

### CODE REVIEW

File: src/masternode/masternode-sync.h [hunk 1]
```diff
@@ -29,6 +31,8 @@ extern CMasternodeSync masternodeSync;
 class CMasternodeSync
 {
 private:
+    CCriticalSection cs;
```
Comment (UdjinM6) [inline src/masternode/masternode-sync.h:34]: Let's maybe
```suggestion
    mutable CCriticalSection cs;
```
and bring back `const`s you dropped?

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
utACK

(dropped "refactor" from the title)

**Review by UdjinM6 — DISMISSED** [review]
ok, unit tests failure (on mac) is related to some UB in ctor (cs internals aren't being fully initialized I think but I don't quite understand why tbh). f04ed6ad89 fixes it for me but there is a similar issue with CHDChain too it seems and I can only reproduce this on my mac 🤷‍♂️  It all looks unrelated to this specific PR, will submit patches in a separate PR.

re-utACK

**Review by xdustinface — CHANGES_REQUESTED** [review]
See 1db3471eb4ed95b2b6b837b5ede836701cf80c2b and 8a581a1dfece77b685b9f3bd5777196476117e5d

**Review by xdustinface — APPROVED** [review]
utACK

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**